### DISCUSSION TITLE
On parameter passing of skywalking in java thread pool

### DISCUSSION THREAD

**Discussion by yuyu414** [apache/skywalking#7621] [Q&A]
If an interface is submitted to the thread pool for execution, how does the traceid information continue to be transmitted without explicit processing?

**Comment by wu-sheng** [comment]
Do you mean trace context? Trace ID should not be propagated separately. If you want to propagate trace context, you need a plugin to call those APIs.

**Reply by yuyu414** [reply]
> Do you mean trace context? Trace ID should not be propagated separately. If you want to propagate trace context, you need a plugin to call those APIs.

I know,What I want to ask is how skywalking is realized.

**Reply by wu-sheng** [reply]
What do you mean `realized`? I was talking about SkyWalking's, not general for other projects.

**Reply by yuyu414** [reply]
![image](https://user-images.githubusercontent.com/25549434/131474395-e4bab2d0-4c47-4a12-9c22-817b526c1957.png)
Because I have encountered some problems, I would like to ask about this implementation principle

**Reply by wu-sheng** [reply]
This depends on which thread generates logs. SkyWalking's Java agent instrument the class and put the trace id(with context) in that thread. Then every log (after plugin works) in the thread, could have logs with trace id.
So, if you want the logs having trace ID, make those codes working under at least one SkyWalking plugin's cover.
